tools/check-file-permissions

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+#**************************************************************************
+#*                                                                        *
+#*                                 OCaml                                  *
+#*                                                                        *
+#*          Sébastien Hinderer, projet Gallium, INRIA Paris               *
+#*                                                                        *
+#*   Copyright 2019 Institut National de Recherche en Informatique et     *
+#*     en Automatique.                                                    *
+#*                                                                        *
+#*   All rights reserved.  This file is distributed under the terms of    *
+#*   the GNU Lesser General Public License version 2.1, with the          *
+#*   special exception on linking described in the file LICENSE.          *
+#*                                                                        *
+#**************************************************************************
+
+# Check whether the repository contains files having an executable permission
+# whereas they should not
+
+# The files for which an executable permission is accepted are:
+# - The scripts, i.e. files starting with #!
+# - boot/ocaml{c,lex}
+# - testsuite/tests/lib-unix/unix-execvpe/subdir/script2, because the test
+#   this file belongs tto requires that it does _not_ start with #!.
+
+ok=true
+
+check_script()
+{
+  if [ x`head -c2 $1` != x'#!' ]; then
```

Comment:
Just to ruin your day: the `-c` option to `head` is not standard (see https://pubs.opengroup.org/onlinepubs/9699919799/utilities/head.html#tag_20_57) and is not implemented in OpenBSD.  (But FreeBSD and MacOS have it.)  So, your script will not run everywhere.  

Possible alternative: `if head -n 1 "$1" | grep -q '^#!' `.